PackManagement.php

```diff
@@ -87,7 +298,8 @@ public function redcap_save_record( $projectID, $recordID, $instrument, $eventID
 		}
 
 		$this->dbGetLock();
```

Comment:
Sorry - probably betraying some of my ignorance again here, but if we've initiated a db lock, and an exception occurs such that execution path drops out of the function before the $this->dbReleaseLock(); - is there a danger the db will be left locked?  Not sure of how the code will react in these circumstances, and what the default exception handling behaviour is here.  I know I've often seen db transactions done with try-catch, and a rollback if all is not well, just wondered if something similar may be beneficial here?

Reply:
The `dbGetLock()` function gets a MySQL named lock as described in the [MySQL locking functions documentation](https://dev.mysql.com/doc/refman/8.0/en/locking-functions.html).
This will wait for the lock with the specific name to become available and there is a timeout in case of an issue releasing a lock. The Pack Management module uses a specific lock name so these locks only affect the module and uses a timeout of 60 seconds.
```php
	public function dbGetLock()
	{
		$this->query( 'DO GET_LOCK(?,60)', [ $GLOBALS['db'] . '.pack_management' ] );
	}
```